Tasks/NuGetToolInstaller/nugetToolInstaller.ts

```diff
@@ -0,0 +1,112 @@
+//import toolLib = require('vsts-task-tool-lib/tool');
+//import taskLib = require('vsts-task-lib/task');
+import {IExecOptions} from "vsts-task-lib/toolrunner";
+
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as taskLib from 'vsts-task-lib/task';
+import * as restm from 'typed-rest-client/RestClient';
+import * as os from 'os';
+import * as path from 'path';
+const uuidV4 = require('uuid/v4');
+
+let osArch: string = os.arch();
+let nugetExePkgName: string = 'nuget.commandline';
+
+async function run() {
+    try {
+        let versionSpec = taskLib.getInput('versionSpec', true);
+        let stableOnly: boolean = taskLib.getBoolInput('stableOnly', false);
+
+        await getNuGet(versionSpec, stableOnly);
+    }
+    catch (error) {
+        console.error('ERR:' + error.message);
+    }
+}
+
+interface INuGetArtifacts {
+    artifacts: INuGetArtifact[]
+}
+
+interface INuGetArtifact{
+    name: string,
+    versions: INuGetVersionInfo[]
+}
+
+interface INuGetVersionInfo {
+    version: string,
+    url: string
+}
+
+async function getNuGet(versionSpec: string, stableOnly: boolean) {
+    toolLib.debug('Trying to get tool from local cache');
+    let localVersions: string[] = toolLib.findLocalToolVersions('NuGet');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+
+    if (version) {
+        console.log('Resolved from tool cache: %s', version);
```

Comment:
May want to add TODO comments to loc. Also, I'm not sure, but the tool lib may be able to handle printing this type of info. I need to look into it.